### ISSUE TITLE
 Compilation warning: 'NSOpenGLContext' on macOS 11

### ISSUE THREAD

**Issue by rageworx** [fltk/fltk#646]
This warning seems to appears on macOS 11 (big-sur) with clang 12.0.5.
Logs is 
```
In file included from src/cocoa_wrapper.mm:2:
In file included from /usr/local/include/FL/x.H:25:
In file included from /usr/local/include/FL/platform.H:39:
/usr/local/include/FL/mac.H:31:8: warning: 'NSOpenGLContext' is deprecated: first deprecated in macOS 10.14 - Please use Metal or MetalKit. [-Wdeprecated-declarations]
extern NSOpenGLContext *fl_mac_glcontext(GLContext rc);
       ^
/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/System/Library/Frameworks/AppKit.framework/Headers/NSOpenGL.h:184:12: note: 'NSOpenGLContext' has been explicitly marked deprecated here
@interface NSOpenGLContext : NSObject <NSLocking>
           ^
1 warning generated.
```
FLTK was latest, and installed to system with sudo.
I'm not sure this is only for me, or something wrong - maybe reason of only for Big-Sur ?
Reporting this for solve future deprecated OpenGL support.

**Comment by MatthiasWM** [comment]
tl;dr Ignore it. We should probably disable the warning.

Apple has introduced Metal in 2014 and deprecated OpenGL shortly after. Yes, it's deprecated, but OpenGL is still fully supported on macOS, and IMHO it will be for may years to come, simply because there is just too much software out there that will never be ported.

Apple is trying really hard to push Metal down our throats. They even wrote their own Metal backend for Evee/Blender, making is blazing fats. But the UI in Blender is still simple old OpenGL.

With FLTK being a cross platform library, I think we should stay away from proprietary libraries like Direct3D and Metal. If we didn't, we would still have to maintain QuickDraw3d, IrisGL, and Glide.

I see our priority in supporting OpenGL3/4/ES. Did Vulkan ever accomplish anything? At least it's cross platform.

**Comment by Albrecht-S** [comment]
I'd like to ask @ManoloFLTK if he can disable this in the macOS code part by some macOS specific means.

If not we could disable specific warnings
1. with `#pragma` (see below)
2. by disabling specific warnings explicitly per file via CMake
3. not sure if the same would (easily) be possible in `Makefile`s

Disabling the warning per commandline, CMake, Makefile has the disadvantage that other warnings in the same file would be undetected. Using `#pragma` might be the best choice because it can be enabled and disabled locally in the specific file.

There's a working example in [src/Fl_x.cxx](https://github.com/fltk/fltk/blob/5e133c1e8813ad19cb64812a4a68d45093a4a217/src/Fl_x.cxx#L1133).

In this case (macOS) I believe we'd need only the `clang` pragma.

**Comment by rageworx** [comment]
I'm not sure , but `#pragma` maybe something like this ?

![image](https://user-images.githubusercontent.com/6948225/211697561-d3cb4bb0-af21-438e-8ac4-4c2de96ee8b0.png)

I'm using this `#pragma` to avoid deprecated-declarations warning cases, but it depends where is right on user-code side, or library-side.

**Comment by ManoloFLTK** [comment]
@rageworx The commit mentionned  above should fix this. Please, close issue if you confirm.

**Comment by rageworx** [comment]
I couldn't test it but believe solved !
Appreciated !

**Comment by rageworx** [comment]
Tested with @Albrecht-S 's suggestion, it was able to ignored precation warning by ...
```
#ifdef __clang__
#pragma clang diagnostic push
#pragma clang diagnostic ignored "-Wdeprecated-declarations"
#endif
```

Actually cannot sure FL/mac.H pragma is working for macOS, line starts 26.
```
 26 #ifdef __OBJC__
 27   @class NSOpenGLContext;
 28 #  ifndef GL_SILENCE_DEPRECATION
 29 #    define GL_SILENCE_DEPRECATION 1
 30 #  endif
```
It seems to ignoring OpenGL deprecation features only, but not related on METAL API suggestion on macOS 11.
I think it may better including clang 1400 should be checked and decide above warning like Fl_x.H in mac.H.

**Comment by rageworx** [comment]
, Or how about notice this Apple macOS deprecation warning issue only happens Big-Sur(11.x) on any OpenGL related document ?
macOS 11 will be buried by new macOS with Apple's compelling, but still some people may remain on macOS 11 like me.
